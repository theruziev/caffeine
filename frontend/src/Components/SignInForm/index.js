import React from 'react'

import useForm from 'react-hook-form'

export default function SignInForm (props) {
  const { register, handleSubmit, errors, setError } = useForm()
  const onSubmit = data => {
    props.onSubmit(data)
  }
  for (let error of props.errors) {
    setError(error.name, error.type, error.message)
  }

  const { submitDisabled, inputsDisabled } = props
  console.log(errors);
  return (

    <form onSubmit={handleSubmit(onSubmit)}>

      <div className='field'>
        <label className='label'>Email</label>
        <div className='control'>
          <input disabled={inputsDisabled} ref={register({ required: true })}
                 className={`input ${errors.username && 'is-danger'}`} name='username' type='text'
                 placeholder='cafe@ephiopia.com'/>
        </div>
        {errors.username && <p className='help is-danger'>{errors.username.message}</p>}
      </div>

      <div className='field'>
        <label className='label'>Password</label>
        <div className='control'>
          <input disabled={inputsDisabled} ref={register({ required: true })}
                 className={`input ${errors.password && 'is-danger'}`} name='password' type='password'
                 placeholder='You password'/>
        </div>
        {errors.password && <p className='help is-danger'>The password is required</p>}

      </div>

      <div className='field is-grouped is-grouped-right'>
        <div className='control'>
          <button className='button is-primary' type='submit' disabled={submitDisabled}>Submit</button>
        </div>
      </div>
    </form>
  )
}
