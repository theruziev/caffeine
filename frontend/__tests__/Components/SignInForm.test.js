import React from 'react'
import SignInForm from '../../src/Components/SignInForm'
import { act, cleanup, fireEvent, render, waitForElement } from '@testing-library/react'

afterEach(cleanup)

describe('<SignInForm />', function () {

  it('snapshot renders', () => {
    const { baseElement } = render(<SignInForm/>)
    expect(baseElement).toMatchSnapshot()
  })

  it('render 2 input', () => {
    const { container } = render(<SignInForm/>)
    expect(container.querySelectorAll('input').length).toEqual(2)
  })

  it('submit empty password and username', async () => {

    const { container, getByText } = render(<SignInForm/>)

    const button = container.querySelector('button')
    fireEvent.click(button)
    const usernameError = await waitForElement(() => getByText('Username is required'))
    const passwordError = await waitForElement(() => getByText('The password is required'))
    expect(usernameError).toHaveClass("is-danger");
    expect(passwordError).toHaveClass("is-danger");
  })


})
