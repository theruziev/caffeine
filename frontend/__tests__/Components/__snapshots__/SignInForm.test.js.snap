// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignInForm /> snapshot renders 1`] = `
<body>
  <div>
    <form>
      <div
        class="field"
      >
        <label
          class="label"
        >
          Email
        </label>
        <div
          class="control"
        >
          <input
            class="input undefined"
            name="username"
            placeholder="cafe@ephiopia.com"
            type="text"
          />
        </div>
      </div>
      <div
        class="field"
      >
        <label
          class="label"
        >
          Password
        </label>
        <div
          class="control"
        >
          <input
            class="input undefined"
            name="password"
            placeholder="You password"
            type="password"
          />
        </div>
      </div>
      <div
        class="field is-grouped is-grouped-right"
      >
        <div
          class="control"
        >
          <button
            class="button is-primary"
            type="submit"
          >
            Submit
          </button>
        </div>
      </div>
    </form>
  </div>
</body>
`;
